### ISSUE TITLE
Warn when no camera exists to render an object

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#1432]
**What problem does this solve or what need does it fill?**

Beginners commonly attempt to get started with adding 2D/3D/UI graphics, forget to add a camera, and then are confused when nothing appears.

**What solution would you like?**

Warn when no camera of the appropriate type exists.

**What alternative(s) have you considered?**

Improve documentation further, especially in the Bevy book.

**Additional context**

The exact timing / details of when this should trigger need some work by others who understand the rendering pipeline more clearly.

We should also be careful not to create unsilenceable warnings in cases where the lack of camera may actually be intended.

**Comment by cart** [comment]
From a comment I left in #1440:

> I think my ideal solution is to make the UiCamera completely optional (and fall back to a default value). Most people don't need control over the UI camera, so theres no real reason to dictate one (other than the fact that its easier to require one because that behavior is consistent with the other render passes)

**Comment by cart** [comment]
The same argument could be made for 2D render passes. But 3d has no "reasonable default", so it might make sense to warn if you try to draw a 3d object when there is no camera.

**Comment by wduminy** [comment]
I am here because this issue is marked as a _good first issue_.  

However, related items indicate that this issue is possibly resolved. 

Is there any more work to be done here?

**Comment by richchurcher** [comment]
Hrm :thinking: The associated PR was closed because it added a check on each frame. The goal is to achieve a warning without using a system. I'll take a poke around, see what the current state is, and whether there might be a semi-obvious solution in 2024.

**Comment by richchurcher** [comment]
Well now you've added `D-Complex` 🙀 😆 

I see UI is already warning about this via a system: 
![image](https://github.com/user-attachments/assets/bfc0b855-cb48-48bb-b435-45f44b9d5bfe)
I'll see if there's anything else to be uncovered that doesn't mess with rendering too badly.

**Comment by alice-i-cecile** [comment]
If it's taken 3 years and multiple attempts to fix I am forced to revise my estimation of difficulty upwards ;)

**Comment by richchurcher** [comment]
Yeah, I'm gonna call defeat on this one until I am more of an "expert" (or less of an ignoramus!)
